sdk/python/touca/cli/_post.py

```diff
@@ -90,13 +94,30 @@ def run(self):
         except ValueError as err:
             print(err, file=sys.stderr)
             return False
-
-        logger.info(f"posting batches one by one")
-        for batchName in batchNames:
-            batchDir = src_dir.joinpath(batchName)
-            logger.info(f"posting {batchDir}")
-            if not _post(batchDir, transport):
-                logger.error(f"failed to post {batchDir}")
-                return False
-        logger.info("posted all result directories")
+ 
+        if (dry_run):
```

Comment:
I'd say we should remove this if else statement and replace it with something like:
```python
if dry_run:
    return dry_run_post(batch_names)
rest of logic in `else` statement.
```
Using a fail fast approach makes this easier to read and understand.